geonode/groups/views.py

```diff
@@ -275,6 +276,14 @@ def getKey(action):
             for action in actions
             if action.action_object and action.action_object.group == self.group.group][:15]
         action_list.extend(context['action_list_maps'])
+        actions = Action.objects.filter(
+            public=True,
+            action_object_content_type__model='document')[:15]
+        context['action_list_documents'] = [
+            action
+            for action in actions
+            if action.action_object and action.action_object.group == self.group.group][:15]
```

Comment:
Shouldn't you filter in the queryset instead ? (not sure, very quick review without looking in depth what the change is about)

Reply:
Nope, in this case, the objects have been stored on "actions" objects. We are basically trying to display all the activities on documents related to a specific group.